File: medinex-backend/src/main/java/com/sistema/rep/servicios/DoctorService.java
package com.sistema.rep.servicios;

import com.sistema.rep.modelo.Servicio;
import com.sistema.rep.modelo.Doctor;

import java.util.List;
import java.util.Set;

public interface DoctorService {

    Doctor agregarDoctor(Doctor doctor);

    Doctor actualizarDoctor(Doctor doctor);

    Set<Doctor> obtenerDoctores();

    Doctor obtenerDoctor(Long doctorId);

    void eliminarDoctor(Long doctorId);

    List<Doctor> listarDoctoresDeUnaServicio(Servicio servicio);

    List<Doctor> obtenerDoctoresActivos();

    List<Doctor> obtenerDoctoresActivosDeUnaServicio(Servicio servicio);
}

File: medinex_frontend/src/app/services/doctor.service.ts
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import baserUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class DoctorService {

  constructor(private http:HttpClient) { }

  public listarCuestionarios(){
    return this.http.get(`${baserUrl}/doctor/`);
  }

  public listarDoctores(){
    return this.http.get(`${baserUrl}/doctor/`);
  }

  public agregarDoctor(doctor:any){
    return this.http.post(`${baserUrl}/doctor/`,doctor);
  }

  public eliminarDoctor(doctorId:any): Observable<any>{
    return this.http.delete(`${baserUrl}/doctor/${doctorId}`, { 
      responseType: 'text',
      observe: 'response'
    });
  }

  public obtenerDoctor(doctorId:any){
    // Temporal: usar endpoint general y filtrar en frontend
    return this.http.get(`${baserUrl}/doctor/`);
  }

  public actualizarDoctor(doctor:any){
    return this.http.put(`${baserUrl}/doctor/`,doctor);
  }

  public listarDoctoresDeUnaServicio(servicioId:any){
    return this.http.get(`${baserUrl}/doctor/servicio/${servicioId}`);
  }

  public obtenerDoctoresActivos(){
    // Temporal: usar endpoint general y filtrar en frontend
    return this.http.get(`${baserUrl}/doctor/`);
  }

  public obtenerDoctoresActivosDeUnaServicio(servicioId:any){
    // Temporal: usar endpoint general y filtrar en frontend
    return this.http.get(`${baserUrl}/doctor/`);
  }
}
